services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3NativeClientConfiguration.java

```diff
@@ -87,11 +112,47 @@ public int maxConcurrency() {
         return maxConcurrency;
     }
 
+    public Executor futureCompletionExecutor() {
+        return futureCompletionExecutor;
+    }
+
+    /**
+     * Finalize which async executor service will be used for the created client. The default async executor
+     * service has at least 8 core threads and can scale up to at least 64 threads when needed depending
+     * on the number of processors available.
+     */
+    private Executor resolveAsyncFutureCompletionExecutor(ClientAsyncConfiguration config) {
+        Supplier<Executor> defaultExecutor = () -> {
+            int processors = Runtime.getRuntime().availableProcessors();
+            int corePoolSize = Math.max(8, processors);
+            int maxPoolSize = Math.max(64, processors * 2);
+            ThreadPoolExecutor executor = new ThreadPoolExecutor(corePoolSize, maxPoolSize,
+                                                                 10, TimeUnit.SECONDS,
+                                                                 new LinkedBlockingQueue<>(1_000),
+                                                                 new ThreadFactoryBuilder()
+                                                                     .threadNamePrefix("sdk-async-response").build());
```

Comment:
Yeah, `sdk-async-response` is used in the future completion thread pool in the Java clients, and I just copied here. I agree that `sdk-async-future-completion` might make more sense, but customers are probably already familiar with this name. Changing the name in transfer manager might confuse them. WDUT?

Reply:
I think as a general rule of thumb, different thread pools should have different naming constructs, since their behavior may vary drastically and it makes them easier to debug.

However, if we feel these thread pools are largely doing the same thing, it may beg the question if it's possible to share a common thread pool between both (e.g., a transfer manager instance must be instantiated from an S3 Java client, which will cause it to inherit the thread pool).

Reply:
Talked offline and decided to keep the same name.